src/runtime/audience-action-flow.ts

```diff
@@ -535,6 +541,53 @@ export class AudienceActionIframeFlow implements AudienceActionFlow {
     this.params_.monetizationFunction?.();
   }
 
+  private async completeAudienceAction() {
```

Comment:
Just curious, are we unable to handle the completion from the controller like the other actions?

Reply:
For rewarded ads completion happens after an ad finishes. We could handle it through the controller, but we would need to keep the iframe open but hide it during the ad. This is a much simpler solution.

Reply:
Makes sense, thanks!